pulumitest/cleanup.go

```diff
@@ -5,6 +5,9 @@ import (
 	"strings"
 )
 
+// We use this interesting contraption of immediately invoked callbacks to
+// eagerly calculate these variables but ensure they're not mutated.
```

Comment:
The "interesting contraption" does nothing to ensure that they are not mutated.

These are mutable variables (as all vars in go are), and can be arbitrarily mutable. This is 100% doable:

```go
skipDestroyOnFailure = func() bool { return true }
```